### PR TITLE
Fix:#881 Insuring we check for and use mkl fftw3.h

### LINKED ISSUE #881: Wrong MKL fftw include path with intel compiler
PR#876 The change from "fftw/fftw3.h" to "fftw3.h" in FindMKL is wrong. The reason of passing on certain machines is the existence of /usr/include/fftw3.h.
We have two options
1, add explict include path MKL/include/fftw
2, revert to the old way just creating a fake fftw.h in build/include.

### PR DESCRIPTION
**Pull request by PDoakORNL**
Closes #881 
The diffs are big but primarily I'm moving around the code so the
intel and non-intel paths are less divergent.
Main fix is to check for ffth3.h in mkl/include/fftw, if it is not
there we do not add that include path.

### CODE REVIEW

File: CMake/FindMKL.cmake [hunk 1]
```diff
@@ -27,20 +43,13 @@ if ( NOT CMAKE_CXX_COMPILER_ID MATCHES "Intel" )
...
-    message(FATAL_ERROR "MKL_INCLUDE_DIRECTORIES not set. \"mkl.h\" not found in MKL_ROOT/(${SUFFIXES})")
-  endif (MKL_INCLUDE_DIRECTORIES-NOTFOUND)
-  message("MKL_INCLUDE_DIRECTORIES: ${MKL_INCLUDE_DIRECTORIES}")
-
   set(MKL_LINKER_FLAGS "-L${MKL_LINK_DIRECTORIES} -Wl,-rpath,${MKL_LINK_DIRECTORIES}")
   set(MKL_LIBRARIES "-lmkl_intel_lp64 -lmkl_sequential -lmkl_core -lpthread -lm -ldl")
 else ( NOT CMAKE_CXX_COMPILER_ID MATCHES "Intel" )
-  # this takes away build control for intel but is convenient
+  # this takes away link control for intel but is convenient
+  # perhaps we should consider dropping it since it will more or less
```
Comment (ye-luo) [inline CMake/FindMKL.cmake]: The non Intel compiler routine hard coded sequential MKL but threaded MKL is better. -mkl picks the threaded one.

Comment (PDoakORNL) [reply CMake/FindMKL.cmake]: Shouldn't we change that then?

Comment (ye-luo) [reply CMake/FindMKL.cmake]: The conclusion from the last discussion was leaving it sequential for non intel compilers.

**Comment by ye-luo** [comment]
Could you verify if EoS works fine now?

**Comment by ye-luo** [comment]
Okay to test

**Comment by PDoakORNL** [comment]
Yes I think I can get on EoS

**Comment by PDoakORNL** [comment]
Verified on EoS with intel by which a mean it builds.

```
Currently Loaded Modulefiles:
  1) modules/3.2.10.5                      17) dvs/2.5_0.9.0-1.0502.2188.1.116.ari
  2) eswrap/1.3.3-1.020200.1280.0          18) alps/5.2.4-2.0502.9774.31.11.ari
  3) switch/1.0-1.0502.60522.1.61.ari      19) rca/1.0.0-2.0502.60530.1.62.ari
  4) gcc/4.8.1                             20) atp/2.0.5
  5) intel/17.0.0.098                      21) PrgEnv-intel/5.2.82
  6) craype-sandybridge                    22) lustredu/1.4
  7) craype-network-aries                  23) xalt/0.7.5
  8) craype/2.5.9                          24) git/2.9.3
  9) cray-mpich/7.5.2                      25) module_msg/0.1
 10) cray-libsci/16.11.1                   26) modulator/1.2.0
 11) udreg/2.3.2-1.0502.10518.2.17.ari     27) hsi/5.0.2.p1
 12) ugni/6.0-1.0502.10863.8.29.ari        28) DefApps
 13) pmi/5.0.13                            29) cmake/2.8.11.2
 14) dmapp/7.0.1-1.0502.11080.8.76.ari     30) cray-hdf5/1.10.0.1
 15) gni-headers/4.0-1.0502.10859.7.8.ari  31) boost/1.60.0
 16) xpmem/0.1-2.0502.64982.5.3.ari

cmake -DENABLE_MKL=1 ..
```

**Pull request merged**